### PR TITLE
Add TTL for refreshing catalog

### PR DESCRIPTION
**Pull request by bitner**
Start of work to add a TTL setting to force refetching the catalog for each collection when the entry for that collection is older than the TTL. When starting up, it always grabs all collections. When hitting the /collections endpoint, it will check each collection and only fetch ones older than the TTL. When hitting any /collections/{collection}/... endpoint, it will check that collection and only request that collection if older than the TTL.

TODO:

- [ ] Add parameter to force refetching.
- [ ] Tests

### CODE REVIEW

File: tipg/db.py [hunk 1]
```diff
@@ -70,15 +70,13 @@ async def __call__(self, conn: asyncpg.Connection):
 
 async def connect_to_db(
     app: FastAPI,
-    settings: Optional[PostgresSettings] = None,
+    settings: Optional[PostgresSettings] = PostgresSettings(),
```
Comment (vincentsarago) [inline tipg/db.py:73]: We should avoid!

File: tipg/dbmodel.py [hunk 2]
```diff
@@ -164,6 +167,7 @@ class Collection(BaseModel):
     geometry_column: Optional[Column]
     datetime_column: Optional[Column]
     parameters: List[Parameter] = []
+    last_catalog_update: datetime = datetime.now()
```
Comment (vincentsarago) [inline tipg/dbmodel.py:170]: default should be set on the post validation step! the issue here is that you'll create a default when we import the dbmodel submodule

File: tipg/logger.py [hunk 3]
```diff
@@ -2,4 +2,5 @@
 
 import logging
 
-logger = logging.getLogger("tipg")
+logger = logging.getLogger("uvicorn")
+logger.setLevel(logging.DEBUG)
```
Comment (vincentsarago) [inline tipg/logger.py:6]: I know this is the easiest way to get the log in uvicorn but we shouldn't do this 😅 

see https://github.com/developmentseed/titiler/issues/276#issuecomment-806250688

File: tipg/db.py (hunk 1 above)
Comment (bitner) [reply tipg/db.py:73]: What is wrong with this?

File: tipg/logger.py (hunk 3 above)
Comment (bitner) [reply tipg/logger.py:6]: It's staying in there on this branch as long as this PR is marked draft!

File: tipg/db.py (hunk 1 above)
Comment (vincentsarago) [reply tipg/db.py:73]: we should not use mutable in function parameters

File: tipg/logger.py (hunk 3 above)
Comment (vincentsarago) [reply tipg/logger.py:6]: 🙈

**Comment by bitner** [comment]
Closing in favor of #68

**Pull request closed without merging**